lib/3llo/remote_server.rb

```diff
@@ -77,6 +77,10 @@ def delete(req_path, req_headers, payload, expected_codes = EXPECTED_CODES)
     private
 
     def build_request_uri(req_path)
+      req_uri = endpoint_url + req_path
+      invalid_uri = (req_uri =~ /\A#{URI::DEFAULT_PARSER.make_regexp}\z/).nil?
+      raise InvalidCommandError.new("Invalid command.") if invalid_uri
+
       URI.parse(endpoint_url + req_path)
```

Comment:
```rb
def build_request_uri(req_path)
  URI.parse(endpoint_url + req_path)
rescue URI::Error
  raise InvalidArgumentError, "invalid command arguments"
end
```

Reply:
Ideally we should introduce a new exception under `RemoteServer` but I think this is fine for now.

Reply:
Well nice!  Such a short and simple way, thank you! I've fixed it, please review again.